source/TextBlade.Core/UserCode/CodeBehindRunner.cs

```diff
@@ -44,19 +42,17 @@ public static void ExecuteLocationCode(Location currentLocation)
         {
             throw new ArgumentException($"{currentLocation.Name} has an empty LocationClass attribute!");
         }
-
-        if (!_classNameToType.ContainsKey(className))
+        if (!_classNameToType.TryGetValue(className, out var type))
         {
-            List<string> allLocationClasses = _classNameToType.Keys.ToList();
+            var allLocationClasses = _classNameToType.Keys.ToList();
             allLocationClasses.Sort();
             throw new ArgumentException($"Looks like {currentLocation.Name} has a LocationClass of {className}, but TextBlade can't find the class. Make sure you add the [LocationCode] attribute to your class, and make sure it's outside the `Content` directory. TextBlade knows about these classes: {String.Join(", ", allLocationClasses)}");
         }
 
         // TODO: document this. We create one instance PER VISIT TO THIS CLASS. And this instance DOES NOT PERSIST.
         // So it has to be ... stateless? I forget the word right this minute. 
         // And, it must have a parameterless constructor. That does whatever it needs to.
-        var type = _classNameToType[className];
         var instance = Activator.CreateInstance(type) as LocationCodeBehind;
-        instance.BeforeShowingLocation(currentLocation);
+        instance?.BeforeShowingLocation(currentLocation);

```

Comment:
If instance is null, that would be an error. We would want to throw here, instead of silently doing nothing.

Reply:
We could definitely throw but I'm assuming here that we could have more than one exposed based type at some point and therefore add code to handle the different ones.

Reply:
Nah, it's not intended to work like that. Developers using TextBlade will specify everything via JSON for the most part. If they need code, the max will be one class per location. So this should always be exactly one. Never 2+ or 0.